src/CatalogDetail/CatalogDetail.js

```diff
@@ -5,39 +5,25 @@ import CatalogSection from '../Section/CatalogSection/CatalogSection'
 import StatisticsSection from '../Section/StatisticsSection/StatisticsSection'
 import OrganizationsSection from '../Section/OrganizationsSection/OrganizationsSection'
 import HarvestsSection from '../Section/HarvestsSection/HarvestsSection'
+import fetchCatalog from '../fetch/fetchCatalog'
+import fetchMetrics from '../fetch/fetchMetrics'
 
 class CatalogDetail extends Component {
   constructor(props) {
     super(props)
     this.state = {catalog: undefined, metrics: undefined}
-    this.getCatalog()
-    this.getMetrics()
   }
 
-  getCatalog() {
-    if (!this.state.catalog) {
-    return fetch(`https://inspire.data.gouv.fr/api/geogw/catalogs/${this.props.params.catalogId}`)
-      .then((response) => response.json())
-      .then((catalog) => {
-        this.setState({catalog})
+  componentWillMount() {
+    return fetchCatalog(this.props.params.catalogId)
```

Comment:
The two `fetch` can be processed in parallel. Consider using `Promise.all`.
You should break this into two component methods: one for Catalog, one for Metrics.